lib/pbench/server/utils.py

```diff
@@ -82,7 +83,8 @@ def quarantine(dest, logger, *files):
             # state. (If it's the associated MD5 file, skip that.)
             if str(afile).endswith(".tar.xz"):
                 try:
-                    Dataset.attach(path=afile, state=States.QUARANTINED)
+                    ds_name = os.path.basename(afile)[: -len(".tar.xz")]
+                    Dataset.attach(name=ds_name, state=States.QUARANTINED)
```

Comment:
Would this be a good place to use `Dataset.stem()`?

Reply:
This was actually where I started dealing with the circular dependency problems that caused me to move it from `Tarball` under `filetree.py` into `Dataset`. And since we're already using `Dataset` here that'd work ... but apparently I missed circling back here.

Reply:
Why was the circularity solved by moving `.stem()` to `Dataset` instead of moving `TARBALL_SUFFIX` to `filetree.py`?

Reply:
The problems come from the fact that `FileTree` is relatively "high level" whereas `Dataset` is "lower" level and in particular is already a requirement for any code dealing with datasets.

Reply:
There's a `Tarball` object, and that's why I'd put the suffix handling in `file_tree.py` when I created it. Indeed, at some point it may make sense to pull `Tarball` out into a separate file, which would simplify some of the dependencies. I'm not sure whether that alone would be sufficient to resolve the immediate problem, and I decided not to try within this PR.

Reply:
Why not make that change independently in a separate PR, get it merged, and then land this one?

Reply:
Because I don't see much value in refactoring the file tree until we know exactly what we want for object store, and it's completely incidental and nearly completely orthogonal to this PR?

Reply:
Except that why does the `Dataset` class have knowledge about the on-disk of the tar ball when the `Tarball` class doesn't.

Reply:
PRs are getting hammered about technical debt of late.  Isn't this technical debt?

Reply:
This isn't "on-disk" structure, though; it's the iconic "agent API convention" of our "data bundle" naming. That's independently enforced at both the agent and server end and retained through artifact access, currently through Apache file sharing and, presumably, by our future internally hosted artifact access when we compute the response header along with shipping off the tarball byte stream.

If we want to change the tarball naming conventions in the future, that's unrelated to the planned transition from `/srv/pbench/archive/<controller>` directory hierarchy to an S3 object URI. So, no, I don't see this as "technical debt" even if in the future we devise a new plan to change this.

It's true that the `Dataset` class is mostly unconcerned with the naming and structural conventions, which was why I'd moved this out to `Tarball` when I created the file tree. I'd (at least slightly) prefer it there, but I don't see it as a big enough architectural issue at this point to mangle the file tree code around before we get serious about object store and know where the refactoring arrow will point. (Since I'm hoping we'll get to that "soon" and it will change a lot, I'd view partial refactoring now as a distraction.)

Reply:
That is certainly one point of view.  It is not worth arguing about.  We'll just change this later.

Reply:
Since most of the infrastructure for `stem()` was already in the `Dataset` class, moving the function there (at least temporarily) makes reasonable sense.  However, it doesn't look to me like `stem()`, or `is_tarball()`, or `TARBALL_SUFFIX` has any real relationship to that class, so perhaps they should (eventually) be hosted somewhere else (and, so, `DatasetBadName` should be disentangled from it, as well).

While these four pieces fit together, I don't see any particular requirement for them to be packaged together in a class.  They could just as well be module-level definitions in some suitable utility module or other place.  I don't know what the details of the circularity problem was, but prying them out of any class altogether would likely help.